sdk/search/azure-search/src/samples/java/com/azure/search/SearchAsyncWithFullyTypedDocumentsExample.java

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.search;
+
+import com.azure.core.http.rest.PagedFluxBase;
+import com.azure.core.util.Configuration;
+import com.azure.search.models.SearchResult;
+import com.azure.search.models.Hotel;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
+
+import java.text.SimpleDateFormat;
+import java.util.TimeZone;
+
+/**
+ * This example shows how to convert a search result into a fully typed object.
+ * In case that the searched document schema is known, the user can convert the result of the search query from a property bag into a fully typed object
+ * In this sample, the deserialization into well known type is done using Jackson ObjectMapper
+ * <p>
+ * This sample is based on the hotels-sample index available to install from the portal.
+ * See https://docs.microsoft.com/en-us/azure/search/search-get-started-portal
+ */
+public class SearchAsyncWithFullyTypedDocumentsExample {
+
+    /**
+     * From the Azure portal, get your Azure Cognitive Search service URL and API key,
+     * and set the values of these environment variables:
+     */
+    private static final String ENDPOINT = Configuration.getGlobalConfiguration().get("AZURE_COGNITIVE_SEARCH_ENDPOINT");
+    private static final String API_KEY = Configuration.getGlobalConfiguration().get("AZURE_COGNITIVE_SEARCH_API_KEY");
+
+    private static final String INDEX_NAME = "hotels-sample-index";
+
+    public static void main(String[] args) {
+        SearchIndexAsyncClient searchClient = new SearchIndexClientBuilder()
+            .endpoint(ENDPOINT)
+            .credential(new SearchApiKeyCredential(API_KEY))
+            .indexName(INDEX_NAME)
+            .buildAsyncClient();
+
+        ObjectMapper objectMapper = new ObjectMapper();
```

Comment:
I thought the SDK was going to include a built-in way of doing this. It seems oddly asymmetric that the SDK handles serialization of typed objects just fine for indexing, but users have to jump through hoops to do the same for deserialization of search results. Is there a better way to do this?

Reply:
@brjohnstmsft it was agreed and discussed in the first  API board review that deserialization will not be part of the SDK, but will be done using external deserialization utility. We initialiy used the jsonwrapper utility from azure-core, but since it is still in dev, and in order to expdite the approval of this PR, it was agreed to seperate the discussion on jsonwrapper to seperate PR. In any stage this can be added to the SDK.

Reply:
@navalev That plan makes sense, with one caveat -- We need to make sure that it's possible in principle for someone (whether us or the customer) to implement this "statically typed" deserialization correctly in all cases. There were some tricky corner cases in the .NET SDK that would have made this layered approach difficult to get right. For example, if a customer defines an index with a string field, then puts an ISO 8601 date-time value in that field, it will be deserialized as `DateTimeOffset` in the "dynamic" case. This would require a conversion back to string if we were then to convert it to a static type. There are similar corner cases for `NaN`/`-INF`/`+INF` for type `Edm.Double`, as well as empty JSON arrays.

As I continue reviewing this PR, I'll look for test coverage that exercises these cases (I know the .NET SDK had it, so I suspect you may have ported it already). As long as those tests exist, it proves in principle that it's possible to implement this correctly, and we're all good. If not, I'll add it to the list of work we need to do after merging the PR, but before we ship. Either way, it won't block the PR.

Reply:
@brjohnstmsft the special corner cases are tested in teh lookup tests (LookupSyncTests and LookupAsyncTests)

Reply:
@brjohnstmsft the special corner cases are tested in the lookup tests (LookupSyncTests and LookupAsyncTests)

Reply:
@brjohnstmsft the special corner cases are tested in teh lookup tests (LookupSyncTests and LookupAsyncTests)